src/Components/Activities/index.js

```diff
@@ -1,11 +1,79 @@
+import { useEffect, useState } from 'react';
 import styles from './activities.module.css';
+import Table from './Table';
+import Form from './Form';
+
+const Activities = () => {
+  const [activities, setActivities] = useState([]);
+  const [showForm, setShowForm] = useState(false);
+  const [selectedActivity, setSelectedActivity] = useState(null);
+
+  const getActivities = async () => {
+    try {
+      const response = await fetch(`${process.env.REACT_APP_API_URL}/api/activity/`);
+      const responseData = await response.json();
+      const data = responseData.data;
+      setActivities(data);
+    } catch (error) {
+      alert('Error fetching activities: ' + error);
+    }
+  };
+
+  useEffect(() => {
+    getActivities();
+  }, []);
+
+  const addActivity = ({ name, description }) => {
```

Comment:
this function is not connected to the DB so if we reload the page we lose all the data, i think the logic the get the data from the forms is correct so only thing missing is to send the data to the DB and work with the response, you also dont need to generate a random id if you connect with the DB